pkg/azure/plugin_integration_test.go

```diff
@@ -360,3 +360,168 @@ func TestMultipleRegionsIntraNamespace(t *testing.T) {
 	require.Nil(t, err)
 	require.True(t, azureConnectivityCheckVM2toVM1)
 }
+
+func TestAttachResourceIntegration(t *testing.T) {
+	namespace := "default"
+	subscriptionId := GetAzureSubscriptionId()
+	resourceGroupName := SetupAzureTesting(subscriptionId, "integration6")
+	defer TeardownAzureTesting(subscriptionId, resourceGroupName, namespace)
+	ctx := context.Background()
+
+	// Set Azure plugin port
+	azureServerPort := 7994
+
+	// Setup orchestrator server
+	orchestratorServerConfig := config.Config{
+		Server: config.Server{
+			Host:    "localhost",
+			Port:    "9092",
+			RpcPort: "9093",
+		},
+		CloudPlugins: []config.CloudPlugin{
+			{
+				Name: utils.AZURE,
+				Host: "localhost",
+				Port: strconv.Itoa(azureServerPort),
+			},
+		},
+		Namespaces: map[string][]config.CloudDeployment{
+			namespace: {
+				{
+					Name:       utils.AZURE,
+					Deployment: getDeploymentUri(subscriptionId, resourceGroupName),
+				},
+			},
+		},
+	}
+	orchestratorServerAddr := orchestratorServerConfig.Server.Host + ":" + orchestratorServerConfig.Server.RpcPort
+	orchestrator.Setup(orchestratorServerConfig, true)
+
+	// Setup Azure plugin server
+	azureServer := Setup(azureServerPort, orchestratorServerAddr)
+
+	vmLocation := "westus"
+	externalVmParameters := GetTestVmParameters(vmLocation)
+	externalVmName := "external-vm"
+	externalVnetName := "external-vnet"
+	externalVmID := getVmUri(subscriptionId, resourceGroupName, externalVmName)
+	resourceIdInfo := ResourceIDInfo{
+		SubscriptionID:    subscriptionId,
+		ResourceGroupName: resourceGroupName,
+		ResourceName:      externalVmName,
+	}
+	externalAddressSpace := "20.9.0.0/24"
+	azureHandler, err := azureServer.setupAzureHandler(resourceIdInfo, namespace)
+	require.NoError(t, err)
+
+	// Create Non-Paraglider Vnet
+	externalVnetParams := CreateVirtualNetworkParameters(ctx, vmLocation, externalVnetName, externalAddressSpace)
```

Comment:
This probably should have been addressed in a previous PR, but could we rename this to `GetVirtualNetworkParameters`? Create makes it sound like it's going to create the VNET.

And also, we shouldn't export this method unless there is a major reason to use this in other packages.